pkg/migrate/migrate.go

```diff
@@ -1146,3 +1296,249 @@ func resetReclaimPolicy(ctx context.Context, w *log.Logger, clientset k8sclient.
 
 	return nil
 }
+
+// buildPVConsumerPod creates a pod spec for consuming a pvc
+func (p *PVMigrator) buildPVConsumerPod(pvcName string) *corev1.Pod {
+	podName := p.podNameOverride
+	if podName == "" {
+		tmp := uuid.New().String()[:5]
+		podName = fmt.Sprintf("pvmigrate-vol-consumer-%s-%s", pvcName, tmp)
+	}
+	if len(podName) > 63 {
+		podName = podName[0:31] + podName[len(podName)-32:]
+	}
+	return &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: "default",
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Volumes: []corev1.Volume{
+				{
+					Name: "tmp",
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "busybox",
+					Image: "busybox",
+					Command: []string{
+						"sleep",
+						"3600",
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/tmpmount",
+							Name:      "tmp",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// buildPVC creates a temporary PVC requesting for 1Mi of storage for a provided storage class name.
+func (p *PVMigrator) buildTmpPVC(pvc corev1.PersistentVolumeClaim, sc string) *corev1.PersistentVolumeClaim {
+	pvcName := p.pvcNameOverride
+	if pvcName == "" {
+		tmp := uuid.New().String()[:5]
+		pvcName = fmt.Sprintf("pvmigrate-claim-%s-%s", pvc.Name, tmp)
+	}
+	if len(pvcName) > 63 {
+		pvcName = pvcName[0:31] + pvcName[len(pvcName)-32:]
+	}
+
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pvcName,
+			Namespace: "default",
```

Comment:
should this be in the same namespace as the src pvc?

Reply:
Not necessarily. We make a copy of the src pvc in the `default` namespace and work within that namespace.

Reply:
we 100% cannot work within the default namespace.

we have to work in the namespaces that the PVCs are in.

Yes, this makes the code a little more complex, but otherwise it breaks this code for environments where 'default' is forbidden which I'd prefer to avoid.